File: apzkr-pzpi-21-11-milennyi-ivan/Task3-WebClient/src/viewModels/FeedSupplyVM.ts
import { timestampToDate } from "../utils/helpers";

// Клас для відображення поставок корму
export class FeedSupplyVM {
    id: number; // Ідентифікатор поставок корму
    amount: number; // Кількість корму в кг
    date: string; // Дата постачання
    storekeeperName: string | null; // Ім'я комірника
    storekeeperSurname: string | null; // Прізвище комірника

    constructor(id: number, amount: number, timestamp: number, storekeeperName: string | null, storekeeperSurname: string | null) {
        this.id = id;
        this.amount = amount / 1000;
        this.storekeeperName = storekeeperName;
        this.storekeeperSurname = storekeeperSurname;
        this.date = timestampToDate(timestamp);
    }
}

File: apzkr-pzpi-21-11-milennyi-ivan/Task1-Server/src/view_models/feed_supply_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

// Структура, що представляє представлення постачання корму (ViewModel)
#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct FeedSupplyVM {
    id: u64, // Ідентифікатор постачання корму
    amount: u64, // Кількість корму
    timestamp: u64, // Час постачання (у вигляді таймстемпу)
    storekeeper_name: Option<String>, // Ім'я комірника (може бути відсутнім)
    storekeeper_surname: Option<String>, // Прізвище комірника (може бути відсутнім)
}